checks.d/kubernetes.py

```diff
@@ -273,11 +330,73 @@ def _update_pods_metrics(self, instance, pods):
                 kind = created_by['reference']['kind']
                 if kind in supported_kinds:
                     controllers_map[created_by['reference']['name']] += 1
-            except KeyError:
+            except (KeyError, ValueError) as e:
+                self.log.debug("Unable to retrieve pod kind for pod %s: %s", pod, e)
                 continue
 
         tags = instance.get('tags', [])
         for ctrl, pod_count in controllers_map.iteritems():
             _tags = tags[:]  # copy base tags
             _tags.append('kube_replication_controller:{0}'.format(ctrl))
             self.publish_gauge(self, NAMESPACE + '.pods.running', pod_count, _tags)
+
+    def _process_events(self, instance, pods_list):
+        """
+        Retrieve a list of events from the kubernetes API.
+
+        At the moment there is no support to select events based on a timestamp query, so we
+        go through the whole list every time. This should be fine for now as events
+        have a TTL of one hour[1] but logic needs to improve as soon as they provide
+        query capabilities or at least pagination, see [2][3].
+
+        [1] https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-apiserver/app/options/options.go#L50
```

Comment:
Nitpick but you should link to a commit, not master. When master changes the link becomes obsolete. As an example, master changed since you opened this PR and now you want to link to L51.